### DISCUSSION TITLE
Master Server Single Point of Failure

### DISCUSSION THREAD

**Discussion by cyberpasta** [Security-Onion-Solutions/securityonion#2863] [Unsupported Versions]
Hellow Team. 
We are an educational institution assessing adopting securityonion. The distributed production architecture has a single point of failure that is the "Master Server".
Is there any solution for this weakness in the 2.3 version?
And if not, what are the recommended controls that we can implement to mitigate the risk of data loss in case of master server went down.

**Comment by cyberpasta** [comment]
I appreciate the team's feedback on this question.

**Comment by petiepooo** [comment]
I saw one suggestion that the master, being it has low resource requirements, could be run on virtual infrastructure.  That assumes you have virtual infrastructure, of course.

**Reply by cyberpasta** [reply]
Many Thanks. Based the the architecture, I suppose having the master as VM with proper backup is the best countermeasure.
I would suggest to include in the documentations some guidance regarding enabling the buffering at the forward notes to cater for an acceptable RPO in case of a master node interruption.
